### PR TITLE
chore: [IOBP-238] IDPAY CODE onboarding flow Xstate and flow basics

### PR DESCRIPTION
**Pull request by forrest57**
## Short description
Addition of Xstate and navigation basics for IDPAY CODE's onboarding flow

## List of changes proposed in this pull request
- addition of xstate:
1. machine definition
2. actions
3. provider
4. services
5. context definition file
6. single guard
-- it is all mocks, so a lot will have to be rewritten, this PR consists in a boilerplate to avoid future merge conflicts --
- Addition of navigator 

## How to test
since no real feature is implemented, there is no way to test  😅 
Read the code and run tsc and prettier, it should be good enough.
once more, keep in mind once again that this is a skeleton made to base the future PRs on on order to have the same folder and naming structures, plus avoid writing boilerplate in every PR

### CODE REVIEW

**Review by Hantex9 — APPROVED** [review]
Since it's a boilerplate to build other features, LGTM!

**Review by mastro993 — CHANGES_REQUESTED** [review]
Will `code` feature need other sub-features? Maybe `ts/features/idpay/code/onboarding` would be better

File: ts/features/idpay/codeOnboarding/xstate/machine.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            src: "generatePin",
+            onDone: {
+              target: "SHOWING_PIN"
+            },
+            onError: {
+              target: "DISPLAYING_ONBOARDING_FAILURE"
+            }
+          }
+        },
+        SHOWING_PIN: {
```
Comment (mastro993) [inline ts/features/idpay/codeOnboarding/xstate/machine.ts:66]:
```suggestion
        DISPLAYING_PIN: {
```

**Review by mastro993 — CHANGES_REQUESTED** [review]

**Comment by mastro993** [comment]
Switched to Redux-Saga https://github.com/pagopa/io-app/pull/4995

**Pull request closed without merging**